qiskit_optimization/translators/ising.py

```diff
@@ -13,17 +13,20 @@
 """Translator between an Ising Hamiltonian and a quadratic program"""
 
 import math
-from typing import Tuple
+from typing import Tuple, Union
 
 import numpy as np
-
 from qiskit.opflow import I, ListOp, OperatorBase, PauliOp, PauliSumOp, SummedOp
-from qiskit.quantum_info import Pauli
+from qiskit.quantum_info import Pauli, SparsePauliOp
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
 from qiskit_optimization.exceptions import QiskitOptimizationError
 from qiskit_optimization.problems.quadratic_program import QuadraticProgram
 
 
-def to_ising(quad_prog: QuadraticProgram) -> Tuple[OperatorBase, float]:
+def to_ising(
+    quad_prog: QuadraticProgram, quantum_info: bool = False
```

Comment:
I imagine this `quantum_info` parameter will get removed once only SparsePauliOp exists. Before then, at some point, we need to transition over to True by default where False is supported but deprecated. So for the release where we introduce this we have it still producing PauliSumOp by default, and somewhere I would think have a warning message saying the default will be changed to SparsePauliOp. When we do that passing False in, to produce a PauliSumOp would be deprecated. We probably should state this intent in the release note. Support both for now defaulting to PauliSumOp, the release after default to SparsePauliOp (deprecating PauliSumOp) and the release after removing that support leaving just SparsePauliOp supported with the flag here gone.

Reply:
Yes, you are right. I plan to remove opflow by 1) allow SparsePauliOp with this PR though the default is opflow, 2) change the default with SparsePauliOp, 3) remove opflow. It also makes sense that the warning message and release note mention the plan.

Reply:
Added a warning if `opflow` is not set explicitly